projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules.php

```diff
@@ -194,12 +202,15 @@ public function create_item( $request ) {
 		 * @param object          $event   The event object.
 		 * @param WP_REST_Request $request The request object.
 		 */
-		do_action( 'jetpack_scheduled_update_created', $id, $event, $request );
+		do_action( 'jetpack_scheduled_update_created', $id, $events[ $id ], $request );
 
 		$event              = wp_get_scheduled_event( Scheduled_Updates::PLUGIN_CRON_HOOK, $plugins, $schedule['timestamp'] );
 		$event->schedule_id = $id;
 		$this->update_additional_fields_for_object( $event, $request );
 
+		// Lock the cache after a successful schedule creation.
+		set_transient( 'jetpack_scheduled_update_created_lock', true, 10 );
```

Comment:
How does it lock the cache?

Reply:
Because if a scheduled event is added at the exact second, it obtains the old cache, and so the cache should be cleared the very first time a new event is scheduled. Despite all the cache clearing done during `update_option` and others.

Reply:
With "lock" I'd expect there to be no changes to the cache, but we're deleting it.

Reply:
If we're consider moving this into a more general place, perhaps something like `pre_schedule_event_clear_cron_cache` ?

Reply:
Renamed.